tools/cli/adminTimers.go

```diff
@@ -114,25 +257,42 @@ func AdminTimers(c *cli.Context) {
 		}
 
 		token = resp.NextPageToken
+		timers = append(timers, resp.Timers...)
+	}
+	return timers
+}
 
-		for _, t := range resp.Timers {
-			if len(domainID) > 0 && t.DomainID != domainID {
-				continue
-			}
+func (fl *fileLoader) Load() []*persistence.TimerTaskInfo {
+	file, err := os.Open(fl.ctx.String(FlagInputFile))
+	if err != nil {
+		ErrorAndExit("cannot open file", err)
+	}
+	var data []*persistence.TimerTaskInfo
+	dec := json.NewDecoder(file)
 
-			if _, ok := taskTypes[t.TaskType]; !ok {
-				continue
+	for {
+		var timer persistence.TimerTaskInfo
+		if err := dec.Decode(&timer); err != nil {
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				break
 			}
+		}
+		data = append(data, &timer)
+	}
+	return data
+}
 
-			data, err := json.Marshal(t)
-			if err != nil {
-				if !skipErrMode {
-					ErrorAndExit("cannot marshal timer to json", err)
-				}
-				fmt.Println(err.Error())
-			} else {
-				fmt.Println(string(data))
-			}
+func (hp *histogramPrinter) Print(timers []*persistence.TimerTaskInfo) error {
+	h := NewHistorgram()
+	for _, t := range timers {
+		if t == nil {
+			continue
 		}
+		h.Add(t.VisibilityTimestamp.Format(hp.timeFormat))
+	}
+
+	if err := h.Print(hp.ctx.Int(FlagShardMultiplier)); err != nil {
```

Comment:
nit: maybe just `return h.Print(hp.ctx.Int(FlagShardMultiplier))` the if does not add anything here.

Reply:
good catch!